### PR TITLE
Feature/fix bugs with modal reservation

### PR DESCRIPTION
**Pull request by Sinisa198**

### CODE REVIEW

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 1]
```diff
@@ -201,6 +209,14 @@ const MealReservation = (): JSX.Element => {
         setConfirmationModalIsOpen(false)
     }
 
+    const isBookingAllowed = (): boolean => {
```
Comment (kiljee) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:212]: move to helper function 
make code more cleaner and easy to read

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 2]
```diff
@@ -482,7 +507,7 @@ const MealReservation = (): JSX.Element => {
                         </div>
                         <div className="mealReservation__openCartContainer__openCartBottom__confirmButtonWrapper">
                             <RegularButton
-                                content="Potvrdi rezervaciju"
+                                content={t('Potvrdi rezervaciju') as string}
                                 style="mealReservation__openCartContainer__openCartBottom__confirmButtonWrapper__confirmButton"
                                 isActive
                                 onClick={() => createOrder()}
```
Comment (kiljee) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:513]: onClick={createOrder}

**Review by kiljee — CHANGES_REQUESTED** [review]

File: moja-kuhinjica-frontend/src/components/modal/disabledReservation/DisabledReservationModal.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                </label>
+                <button
+                    className="modalContainer__formContainer__formDiv__formTitle__formButton"
+                    onClick={closeModal}
+                >
+                    {t(buttonText)}
+                </button>
+            </div>
+        )
+    }
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/components/modal/disabledReservation/DisabledReservationModal.tsx:42]: remove this function, use conditional rendering

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 4]
```diff
@@ -40,11 +42,13 @@ import { routes } from '../../constants/constants'
...
 const HEADER_TYPE = 'red'
 const INITIAL_MEAL_AMOUNT = 1
+const DISABLED_MESSAGE = 'Ne možete da rezervišete posle 10 ujutru'
 const RESERVATION_SUCCESS = 'Rezervacija je uspešna'
 const RESERVATION_FAIL = 'Neuspešna rezervacija'
 const RESERVATION_SUCCESS_MESSAGE =
     'Vaša rezervacija je sačuvana. Možete je pogledati na stranici'
-const EMPTY_CART_MESSAGE = 'Vaša korpa je prazna, ukoliko želite da pogledate svoje rezervacije možete otići na stranicu'
+const EMPTY_CART_MESSAGE =
+    'Vaša korpa je prazna, ukoliko želite da pogledate svoje rezervacije možete otići na stranicu'
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:51]: move these constants to different file

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 5]
```diff
@@ -136,6 +142,10 @@ const MealReservation = (): JSX.Element => {
             })
     }
     const addToCart = (meal: IMeal): void => {
+        if (!isBookingAllowed(activeDate)) {
+            return setShowDisabledReservation(true)
+        }
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:147]: inline if

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 6]
```diff
@@ -237,6 +247,13 @@ const MealReservation = (): JSX.Element => {
                 }}
                 buttonText="OK"
             />
+            <DisabledReservationModal
+                modalIsOpen={showDisabledReservation}
+                closeModal={() => setShowDisabledReservation(!true)}
+                title={DISABLED_MESSAGE}
+                buttonText={'OK'}
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:254]: buttonText='OK'

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 7]
```diff
@@ -275,11 +292,11 @@ const MealReservation = (): JSX.Element => {
...
                                         active={active === activeTabIndex}
                                         onClick={() => {
-                                            setDeyOfWeek(date.day())
-                                            setActive(activeTabIndex)
                                             if (cartItems.length) {
                                                 handleTabClickWithCartItems()
                                             } else {
+                                                setDeyOfWeek(date.day())
+                                                setActive(activeTabIndex)
                                                 setActiveDate(day.date)
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:300]: dont use `if - else`

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 8]
```diff
@@ -342,12 +359,17 @@ const MealReservation = (): JSX.Element => {
...
+                                        <span
                                             className="mealReservation__container__menuDiv__cartContainer__cartWrapper__emptyCartDiv__emptyCartLabel__emptySpan"
-                                            onClick={() => router.push(routes.MY_RESERVATIONS_PAGE)}    
-                                        > 
+                                            onClick={() =>
+                                                router.push(
+                                                    routes.MY_RESERVATIONS_PAGE
+                                                )
+                                            }
+                                        >
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:372]: use Link

**Review by vladavoX — APPROVED** [review]

**Review by kiljee — APPROVED** [review]

**Pull request merged**